frb_rust/src/support.rs

```diff
@@ -50,3 +50,60 @@ pub struct WireSyncReturnStruct {
     pub len: i32,
     pub success: bool,
 }
+
+/// Macro for implementing [`From<Primitive>`] for [`WireSyncReturnStruct`].
+/// This conversion won't fail.
+macro_rules! primitive_to_sync_return {
+    ($($t:ty),+) => {
+        $(impl From<$t> for WireSyncReturnStruct {
+            fn from(data: $t) -> Self {
+                let bytes = data.to_be_bytes().to_vec();
+                let (ptr, len) = into_leak_vec_ptr(bytes);
+                WireSyncReturnStruct {
+                    ptr,
+                    len,
+                    success: true,
+                }
+            }
+        })*
+    }
+}
+
+// For simple types, use macro to implement [`From`] trait.
+primitive_to_sync_return!(u8, i8, u16, i16, u32, i32, u64, i64, f32, f64);
+
+/// Bool will be converted to u8 where 0 stands for false and 1 stands for true.
+impl From<bool> for WireSyncReturnStruct {
+    fn from(data: bool) -> Self {
+        let bytes = if data { 1_u8 } else { 0_u8 }.to_be_bytes().to_vec();
+        let (ptr, len) = into_leak_vec_ptr(bytes);
+        WireSyncReturnStruct {
+            ptr,
+            len,
+            success: true,
+        }
+    }
+}
+
+impl From<String> for WireSyncReturnStruct {
+    fn from(data: String) -> Self {
+        let bytes = data.as_bytes().to_vec();
+        let (ptr, len) = into_leak_vec_ptr(bytes);
+        WireSyncReturnStruct {
+            ptr,
+            len,
+            success: true,
+        }
+    }
+}
```

Comment:
Similarly, using the idea of delegating, we can do it in 1 line:

```
data.as_bytes().to_vec().into()
```

And done